Plugin/LSP/LSPOutlineView.cpp

```diff
@@ -0,0 +1,1160 @@
+    m_treeCtrl->SetItemTextColour(node, style.colour);
+    
+    // recursively add children
+    for(const auto& child : symbol.GetChildren()) {
+        AddDocumentSymbolRec(node, child, lexer);
+    }    
+}
+
+
+void LSPOutlineView::DoInitialiseDocumentSymbol()
+{   
+    // show only tree and toolbar
+    m_msgPanel->Hide();
+    m_msgIndicator->Stop();
+    m_terminalViewCtrl->Hide();
+    m_treeCtrl->Show();
+    
+    // enable all tools
+    m_toolbar->EnableTool(ID_TOOL_EXPAND_ALL, true);
+    m_toolbar->EnableTool(ID_TOOL_COLLAPSE_ALL, true);
+    m_toolbar->EnableTool(ID_TOOL_DETAILS, m_mode == MODE::DOCUMENT_SYMBOL);
+    m_toolbar->EnableTool(ID_TOOL_KIND, true);
+    m_toolbar->EnableTool(ID_TOOL_MENU, true);
+    m_sortOptions->Enable();
+    
+    auto lexer = ColoursAndFontsManager::Get().GetLexer("python");
+    m_treeCtrl->DeleteAllItems();
+    if (m_textCtrlFilter)
+        m_textCtrlFilter->Clear();
+
+    m_treeCtrl->Begin();
+    clColours colours;
+    colours.FromLexer(lexer);
+    m_treeCtrl->SetColours(colours);
+    
+    wxFont font = lexer->GetFontForStyle(0, m_treeCtrl);
+    if(m_style & STYLE_SMALL_FONT)
+        font.Scale(0.8f);
+    m_treeCtrl->SetDefaultFont(font);
+    
+    wxTreeItemId node = m_treeCtrl->AddRoot("Root");
+    for (const DocumentSymbol& ds : m_documentSymbols) {
+        AddDocumentSymbolRec(node, ds, lexer.get());
+    }
+    
+    m_treeCtrl->Commit();
+    
+    // expand or collapse the tree according to config setting
+    InitExpandedState();
+    
+    m_treeCtrl->Unbind(wxEVT_TREE_ITEM_EXPANDED, &LSPOutlineView::OnTreeItemExpanded, this);  
+    m_treeCtrl->Unbind(wxEVT_TREE_ITEM_COLLAPSED, &LSPOutlineView::OnTreeItemCollapsed, this);  
+    
+    // items in m_treeState are DocumentSymbols wrapped in wxTreeItemId
+    // the tree cannot directly use these, we have to find the corresponding row recursively first
+    // easier to iterate the whole tree and check if each item is in our m_treeState
+    RestoreTreeStateRec(m_treeCtrl->GetRootItem());
+    
+    m_treeCtrl->Bind(wxEVT_TREE_ITEM_EXPANDED, &LSPOutlineView::OnTreeItemExpanded, this);  
+    m_treeCtrl->Bind(wxEVT_TREE_ITEM_COLLAPSED, &LSPOutlineView::OnTreeItemCollapsed, this);  
+        
+    Layout();
+    if (m_textCtrlFilter)
+        m_textCtrlFilter->CallAfter(&wxTextCtrl::SetFocus);
+}
+
+void LSPOutlineView::RestoreTreeStateRec(wxTreeItemId parent)
+{
+    wxTreeItemIdValue cookie;
+    wxTreeItemId child = m_treeCtrl->GetFirstChild(parent, cookie);
+    while (child.IsOk()) {
+        // get the DocumentSymbol
+        auto* data = m_treeCtrl->GetItemData(child);
+        
+        if (data->GetId() == m_treeState.selectedNode) {
+            m_treeCtrl->SelectItem(child); 
+            m_treeCtrl->EnsureVisible(child);           
+        }
+        
+        auto it2 = std::find(m_treeState.collapsedNodes.begin(), m_treeState.collapsedNodes.end(), data->GetId());
```

Comment:
```suggestion
        auto it2 = std::ranges::find(m_treeState.collapsedNodes, data->GetId());
```
BTW strange to begin with `it2`, then `it`.

```suggestion
        if (std::ranges::find(m_treeState.collapsedNodes, data->GetId()) != m_treeState.collapsedNodes.end())
```
would avoid extra variable, as you dont use it BTW.